### PR TITLE
feat: Make lib work with shadow dom - use composedPath

### PR DESCRIPTION
**Pull request by newying61**
@Pomax , @cameronbraid
Hi, this is a possible solution for making lib work inside shadowDOM.

I added tests.js which renders react app inside a shadow DOM.

Got an error when build the project. Will look into it later.

### CODE REVIEW

**Comment by mbIkola** [comment]
> Got an error when build the project. Will look into it later.
`warnings` option added to a wrong place. 
Following patch  should fix your build:
```diff
diff --git a/rollup.config.js b/rollup.config.js
index 9f59da4..dcc9e01 100644
--- a/rollup.config.js
+++ b/rollup.config.js
@@ -39,9 +39,9 @@ const prodUmdConfig = mergeAll([
         compress: {
           pure_getters: true,
           unsafe: true,
-          unsafe_comps: true,
-          warnings: false,
+          unsafe_comps: true
         },
+        warnings: true
       })
     ),
   },
```
Details: UglifyJS [compress options](https://github.com/mishoo/UglifyJS2/blob/master/README.md#compress-options) and [minify options](https://github.com/mishoo/UglifyJS2/blob/master/README.md#minify-options)

**Comment by newying61** [comment]
@mbIkola 
Thanks for the advice. Will update next week.

**Comment by newying61** [comment]
@mbIkola
Thanks. Build succeeded. Tests all passed.

By the way, any reason using closed mode shadow DOM?
Like avoid abusing .shadowRoot?
There is really little benefit we can get from { mode: 'closed' }.

File: test/browser/test3.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          React.createElement('button', {
+            onClick: e => this.state.hideToolbox && this.show(e),
+            children: 'show text inside shadow DOM',
+          }),
+          this.state.hideToolbox ? null : React.createElement(Popup, { hide: e => this.hide(e) }),
+        ],
+      });
+    }
+    show() {
+      console.log('test3 - show');
```
Comment (MateuszKrazek) [inline test/browser/test3.js]: console log with test

File: test/browser/test3.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      this.setState({ hideToolbox: false });
+    }
+    hide() {
+      console.log('test3 - hide');
```
Comment (MateuszKrazek) [inline test/browser/test3.js]: console log with test

**Comment by fedelocci** [comment]
Sorry guys, there's news for this PR. Thanks

**Comment by newying61** [comment]
@fedelocci 
thanks.

**Comment by dvasconcelos1** [comment]
Come on guys, let's merge this...

**Comment by sagyos** [comment]
Hi guys are you going to merge this soon? @MateuszKrazek

**Comment by snasirca** [comment]
Please merge this soon as we need it as well for `react-datepicker`

**Comment by Pomax** [comment]
closing due to inactivity - mine, mostly, and unless someone is willing to fund me working on this again, that's unlikely to change.

**Pull request closed without merging**